WordPress/gutenberg
# Navigation Editor - Enable save button when change made + add indicators to changed elements

## Description

Advances https://github.com/WordPress/gutenberg/issues/30322
Closes https://github.com/WordPress/gutenberg/issues/30391

This PR aims to bring the functionality of enabling the save button only when there are unsaved changes. Also, visible indicators are added to inform users which elements are unsaved.

I have added a mechanism that traces the changes, thanks to which it will be easier to implement https://github.com/WordPress/gutenberg/issues/30423 with information which data is unsaved and will be lost.

Add new pages is not yet saved on `save` button - there is a separate issue for this https://github.com/WordPress/gutenberg/issues/30425

Currently, the dot indicator appears next to `Name` label and not inside the input,
when it comes to Theme location, if a user makes any change the dot will appear next to the first checkbox.

In order to move the dot inside the input, then we would need to style it in a brutal way, placing it absolutely from a container containing entire TextControl elements (or with float, which would be an even worse solution). This would be not a future-proof solution and I don't like it. 
Alternatively, we can make a change inside a TextControl component, so the dot can be styled there.
This change can be added in a separate PR.

If we would like to make changes inside components, we would be also capable of adding dot indicator for each theme location checkbox separately. Without changing ChexbkoxControl the only way is to add a dot as a Header (see issue: https://github.com/WordPress/gutenberg/pull/30037) but it will move the entire Checkbox to the right what would look weird.

There is an issue with a `Tooltip` element - when it is added to a disabled button it adds `,` at the end also makes a button clickable. I have added [issue](https://github.com/WordPress/gutenberg/issues/30422) about this and will also investigate it soon.

## How has this been tested?



1. Navigate to Navigation Editor
2. Select menu (or create one)
3. Make any change in the navbar
4. See the button changes from disabled to enabled and dot indicators are shown in places where a change has been made

## Screenshots 
![image](https://user-images.githubusercontent.com/8398557/113191235-94250500-925d-11eb-856d-3cd29279fa2a.png)


## Types of changes





## Checklist:
